build.sbt

```diff
@@ -155,6 +156,32 @@ lazy val semanticdbScalacPlugin = project
   )
   .dependsOn(semanticdbScalacCore)
 
+lazy val semanticdbJavacPlugin = project
+  .in(file("semanticdb/javac"))
+  .settings(
+    moduleName := "semanticdb-javac",
+    description := "Javac compiler plugin that generates SemanticDB on compile",
+    publishableSettings,
+    mergeSettings,
+    fullCrossVersionSettings,
+    pomPostProcess := { node =>
+      new RuleTransformer(new RewriteRule {
+        private def isAbsorbedDependency(node: XmlNode): Boolean = {
+          def isArtifactId(node: XmlNode, fn: String => Boolean) =
+            node.label == "artifactId" && fn(node.text)
+          node.label == "dependency" && node.child.exists(child =>
+            isArtifactId(child, _.startsWith("semanticdb-javac")))
+        }
+        override def transform(node: XmlNode): XmlNodeSeq = node match {
+          case e: Elem if isAbsorbedDependency(node) =>
+            Comment("the dependency that was here has been absorbed via sbt-assembly")
+          case _ => node
+        }
+      }).transform(node).head
+    }
+  )
+  .dependsOn(scalametaJVM, metacp)
```

Comment:
Also why is the dependency on `metacp` necessary?

Reply:
`SemanticdbIndex` is used